src/ngx_http_lua_headers.c

```diff
@@ -1345,22 +1349,51 @@ ngx_http_lua_ffi_req_header_set_single_value(ngx_http_request_t *r,
 
     k.len = key_len;
 
-    if (value_len == 0) {
-        v.data = NULL;
-        v.len = 0;
+    if (mvals) {
+        if (mvals_len > 0) {
+            for (i = 0; i < mvals_len; i++) {
+                p = mvals[i].data;
+                len = mvals[i].len;
 
-    } else {
+                v.data = ngx_palloc(r->pool, len + 1);
+                if (v.data == NULL) {
+                    return NGX_ERROR;
+                }
+
+                ngx_memcpy(v.data, p, len);
+                v.data[len] = '\0';
+                v.len = len;
+
+                if (ngx_http_lua_set_input_header(r, k, v, override)
+                    != NGX_OK)
+                {
+                    return NGX_ERROR;
```

Comment:
`ngx_http_lua_ffi_set_resp_header` has a slightly different implementation. I believe this implementation should closely mirror it. Additionally, `ngx_http_lua_ffi_set_resp_header` also sets error messages on "failed to set header" or "no memory" errors. I think that we should do the same here.